cli/admin/src/main/java/io/pravega/cli/admin/controller/metadata/ControllerMetadataCommand.java

```diff
@@ -0,0 +1,145 @@
+import static io.pravega.cli.admin.serializers.controller.ControllerMetadataSerializer.EMPTY;
+
+public abstract class ControllerMetadataCommand extends ControllerCommand {
+    static final String COMPONENT = "controller-metadata";
+    static final ControllerKeySerializer KEY_SERIALIZER = new ControllerKeySerializer();
+
+    protected final GrpcAuthHelper authHelper;
+
+    /**
+     * Creates a new instance of the Command class.
+     *
+     * @param args The arguments for the command.
+     */
+    public ControllerMetadataCommand(CommandArgs args) {
+        super(args);
+
+        authHelper = new GrpcAuthHelper(true,
+                "secret",
+                600);
+    }
+
+    /**
+     * Method to get the entry corresponding to the provided key in the table.
+     *
+     * @param tableName          The name of the table.
+     * @param key                The key.
+     * @param segmentStoreHost   The address of the segment store instance.
+     * @param serializer         The valid {@link ControllerMetadataSerializer}.
+     * @param adminSegmentHelper An instance of {@link AdminSegmentHelper}.
+     * @return The object in the queried table segment entry.
+     */
+    Object getTableEntry(String tableName, String key, String segmentStoreHost,
+                         ControllerMetadataSerializer serializer, AdminSegmentHelper adminSegmentHelper) {
+        ByteArraySegment serializedKey = new ByteArraySegment(KEY_SERIALIZER.serialize(key));
+
+        List<TableSegmentEntry> entryList = getIfTableExists(adminSegmentHelper.readTable(tableName,
+                new PravegaNodeUri(segmentStoreHost, getServiceConfig().getAdminGatewayPort()),
+                Collections.singletonList(TableSegmentKey.unversioned(serializedKey.getCopy())),
+                authHelper.retrieveMasterToken(), 0L), tableName);
+        if (entryList == null) {
+            return null;
+        }
+
+        if (entryList.get(0).getKey().getVersion().equals(TableSegmentKeyVersion.NOT_EXISTS)) {
+            output(String.format("Key not found: %s", key));
+            return null;
+        }
+        return serializer.deserialize(getByteBuffer(entryList.get(0).getValue()));
+    }
+
+    /**
+     * Method to safely complete a table query.
+     *
+     * @param future    The CompletableFuture containing the table query.
+     * @param tableName The table name.
+     * @param <T>       Type of the result of the table query.
+     * @return The result of future.join().
+     */
+    <T> T getIfTableExists(CompletableFuture<T> future, String tableName) {
+        return Futures.getAndHandleExceptions(future, t -> {
+           if (t instanceof WireCommandFailedException) {
+               if (((WireCommandFailedException) t).getReason().equals(WireCommandFailedException.Reason.SegmentDoesNotExist)) {
+                   output(String.format("Table not found: %s", tableName));
+                   return null;
```

Comment:
A readTable or getEntry API can fail because of :
1. Table does not exist - which is captured here by - WireCommandFailedException.Reason.SegmentDoesNotExist
2. Key is not found in the table - WireCommandFailedException.Reason.TableKeyDoesNotExist ( you need to handle this as well) --> - print a suitable message
3. Some other issue 
      - AuthFailed - WireCommandFailedException.Reason.AuthFailed --> - print a suitable message
      - WrongHost - WireCommandFailedException.Reason.UnknownHost --> - print a suitable message
      - Some other exception occured - A catch all that tells the user something went wrong and we cound not process your request pleas try again later.....
Need to add code for (2) and (3) here.

Reply:
Instead of using `getAndHandleException()` it may be better to just have an exceptionally block where you can check for the various Exception types and print the appropriate message... something like...
```
return future.exceptionally(ex -> {
            if (isExpected.test(Exceptions.unwrap(ex))) {
                /// print some message
            } 
           return empty value 
        });
```

Reply:
Used `getThrowingException` instead and each exception is handled separately without throwing it